golang.go

```diff
@@ -443,25 +451,58 @@ func (m *GolangConfigTest) testDocker(ctx context.Context, packages ...string) e
 			Consistency: "delegated",
 		})
 
-	gocmd := m.builcCmd(packages...)
+	gocmd := m.buildCmd(packages...)
 
 	return trace.Wrap(cmd.Run(ctx, m.BuildContainer, gocmd[0], gocmd[1:]...))
 }
 
 func (m *GolangConfigTest) testLocal(ctx context.Context, packages ...string) error {
-	gocmd := m.builcCmd(packages...)
+	gocmd := m.buildCmd(packages...)
 	_, err := m.target.Exec().SetEnvs(m.Env).Run(ctx, gocmd[0], gocmd[1:]...)
 	return trace.Wrap(err)
 }
 
-func (m *GolangConfigTest) builcCmd(packages ...string) []string {
+func (m *GolangConfigTest) buildCmd(packages ...string) []string {
 	cmd := []string{"go", "test"}
 
 	cmd = append(cmd, m.GolangConfigCommon.genFlags()...)
+	if len(m.coverProfile) != 0 {
+		cmd = append(cmd, "-cover", "-coverprofile", m.coverProfilePath())
+	}
+	count := m.count
+	if !m.cacheResults && count == 0 {
+		count = 1
+	}
+	if count != 0 {
+		cmd = append(cmd, "-count", fmt.Sprint(count))
+	}
 
 	cmd = append(cmd, packages...)
 
-	return cmd
+	if len(m.coverProfile) == 0 {
+		return cmd
+	}
+
+	return []string{
+		"bash", "-exc",
```

Comment:
hmmm... do we prefer to build a pipeline in bash, or just exec / log the commands independently of eachother?

Reply:
Please see 0fe1c91 for changes addressing this.